pkg/expression/constant.go

```diff
@@ -236,9 +237,22 @@ func (c *Constant) Traverse(action TraverseAction) Expression {
 	return action.Transform(c)
 }
 
+// EvalWithInnerCtx evaluates expression with inner ctx.
+// Deprecated: This function is only used during refactoring, please do not use it in new code.
+// TODO: remove this method after refactoring.
+func (c *Constant) EvalWithInnerCtx(row chunk.Row) (types.Datum, error) {
+	var ctx sessionctx.Context
+	if c.DeferredExpr != nil {
+		if sf, sfOk := c.DeferredExpr.(*ScalarFunction); sfOk {
+			ctx = sf.GetCtx()
+		}
+	}
+	return c.Eval(ctx, row)
+}
```

Comment:
`EvalWithInnerCtx` will use the the context in the `ScalarFunction` as the input parameter for `Eval`. If `c.DefferedExpr` is nil, the ctx will also be nil, don't worry because `ctx` will not be used in this case

Reply:
`EvalWithInnerCtx` will be removed after we replaces all `EvalWithInnerCtx` with `Eval` in the later PRs